src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -240,6 +428,32 @@ EvaluateKernel(const arma::vec& query, const arma::vec& reference) const
   return kernel.Evaluate(metric.Evaluate(query, reference));
 }
 
+template<typename MetricType, typename KernelType, typename TreeType>
+double KDERules<MetricType, KernelType, TreeType>::
+CalculateAlpha(TreeType* node)
+{
+  KDEStat& stat = node->Stat();
+  if (std::abs(stat.MCBeta() - mcBeta) > DBL_EPSILON)
+  {
+    TreeType* parent = node->Parent();
+    if (parent == NULL)
+    {
+      // Base case.
+      stat.MCAlpha() = mcBeta;
+    }
+    else
+    {
+      // Recurse.
+      stat.MCAlpha() = CalculateAlpha(parent) / parent->NumChildren();
```

Comment:
I think that if we can assume that a parent node is always visited before a child node (which is true for all mlpack traversals), then we can just do `stat.MCAlpha() = parent->Stat().MCAlpha() / parent->NumChildren()`.  Let me know if I overlooked something though.

Reply:
That's right, but I have issues with Binary trees and Octrees.

Both their traversals work in the way you mentioned but they don't call the ```Score()``` method for the root node, therefore ```rootNode.Stat().MCAlpha() != mcBeta``` instead (if the tree has never been used) will be ```rootNode.Stat().MCAlpha() = 0```.

I can see a few ways of making this work, either:

1. Leave this method as it is.
2. Manually assigning ```rootNode.Stat().MCAlpha()``` before calling the traversal (I think this one would be interesting).
3. Calling ```Score()``` in the root nodes of the trees' traversals (although I tried this one and didn't seem to work very well, I need to have a deeper look).

Reply:
Huh, I see.  I would imagine that calling `Score()` on the root nodes wouldn't be any problem though---it presumably shouldn't do any pruning.  If you want I can help dig deeper into that, just let me know.

Reply:
The traversals issue should be solved with 3554b2cd9e740432f01a902e0b56d0bca6085e7b and 1dc6df50e21e7d9c95b9a7712d16b24346a9b5bb. ```CalculateAlpha()``` has been improved in a677fa3ce88ac2efb0795b91c6e3e6a26a3a7f00 :)